app/components/gh-image-uploader.js

```diff
@@ -177,7 +192,9 @@ export default Component.extend({
     generateRequest() {
         let ajax = this.get('ajax');
         let formData = this.get('formData');
-        let url = `${ghostPaths().apiRoot}/uploads/`;
+        let isIcon = this.get('isIcon');
+        // CASE: we want to upload an icon and we have to POST it to a different endpoint, expecially for icons
+        let url = isIcon ? `${ghostPaths().apiRoot}/uploads/icons/` : `${ghostPaths().apiRoot}/uploads/`;
```

Comment:
Rather than build domain knowledge into a generic component it would be better to add an `uploadURL` attribute so that it can be re-used anywhere in the app, keeping the domain knowledge in the higher level components/controllers. If you follow the same `didReceiveAttrs` pattern as the accept/extensions attributes that would be 👍